app/models/challenge_models/gift_exchange.rb

```diff
@@ -80,4 +80,21 @@ def user_allowed_to_see_requests_summary?(user)
   def user_allowed_to_see_prompt?(user, prompt)
     self.collection.user_is_maintainer?(user) || prompt.pseud.user == user
   end
+
+  # reindex collection after creation, deletion, and certain attribute updates
```

Comment:
Can you move this to ChallengeCore, since it's shared by both types of challenges?

Reply:
Is there a trick to working with this shared module. I couldn't get it to recognize the activerecord methods necessary for this. I kept getting: undefined method `saved_changes' for #<Class:0x000000000804d960>`